docs/tutorial-distribution.md

```diff
@@ -0,0 +1,91 @@
+# Introduction
+
+A FAW distribution is a specific configuration of the FAW. It is usually associated with a specific format/corpus; for example, the FAW repository contains a small set of distributions for file formats including one for PDFs.
+
+In this tutorial, we will look at creating a new CSV distribution from scratch. Additional information can also be found by looking at how the example distributions have been put together; in particular, the corresponding `config.json5` files often contain comments about configuring/using particular features of the FAW.
+
+# Getting Started
+
+Distributions are typically created as separate folders within the FAW repository. These folders consist of a top-level `config.json5` file and various plugins. While it is conventionally organized with each plugin having its own subfolder and configuration file, these files can be merged with the main configuration. They can also define the process for building and installing dependencies required during runtime, which can range from simple system package installations to complex multi-stage builds. Moreover, the distribution image can be tailored to include only essential files and binaries, ensuring a clear separation between the plugin's build process and the necessary runtime artifacts.
+
+A simple top-level configuration file for a simple CSV parser/validator looks like the below:
+
+```json5
+{
+  // The name of the docker image to be produced.
+  name: 'galois-workbench-csv',
+  
+  parsers: {
+    csvvalidator: {
+      exec: ['csv_validator', '<inputFile>'],
+      version: '0.1',
+      parse: {
+        type: 'regex-counter',
+        version: '1',
+        stdout: {
+          '(.*)': {
+            nameGroup: 1,
+          },
+        },
+      },
+    },
+  },
+  decision_default: {file: 'dsl.txt'},
+  decision_views: {
+  },
+
+  file_detail_views: {
+  },
+  build: {
+    stages: {
+      base: {
+        from: 'ubuntu:22.04',
+        commands: [
+          'RUN apt-get update && apt-get install -y python3 python3-pip'
+        ]
+      },
+      csvvalidator: {
+        commands: [
+          'COPY ./csv/csv_validator.py /tmp/csv_validator.py'
+        ],
+        copy_output: {
+          '/tmp/csv_validator.py': '/usr/bin/csv_validator'
+        },
+      },
+    },
+  },
+}
+```
+
+In the next few sections we will describe relevant portions of the configuration.
```

Comment:
This file would be a great place to move/duplicate the inline documentation in the PDF `config.json5`. If you don't plan to do so in this PR, make sure to link to that example here.

Reply:
I have added links to relevant configuration files (instead of duplicating the comments), mostly because it may make more sense to read the field descriptions in context and there is a single authoritative source for what each field means. Please take a look at commit [994553a](https://github.com/GaloisInc/FAW/commit/994553a02c74c2d643c114b4b5db54edd67397d7).